boa/src/builtins/value/val_type.rs

```diff
@@ -28,6 +29,7 @@ impl Type {
             Self::Function => "function",
             Self::Object => "object",
             Self::BigInt => "bigint",
+            Self::Date => "date",
```

Comment:
We should remove this, there isn't a `"date"` primitive.

Reply:
That resolves my above concerns, I have removed `Date::Value` entirely and changed `to_json` to match the spec (it must call `toJSON()` if that function exists), as well as `neg` calling `to_numeric_number`.